api/tests/integration/infrastructure/repositories/target-profile-share-repository_test.js

```diff
@@ -1,39 +1,81 @@
-      targetProfileB = databaseBuilder.factory.buildTargetProfile();
-      targetProfileC = databaseBuilder.factory.buildTargetProfile();
+      organizationId = databaseBuilder.factory.buildOrganization().id;
+      targetProfileIdA = databaseBuilder.factory.buildTargetProfile().id;
+      targetProfileIdB = databaseBuilder.factory.buildTargetProfile().id;
+      targetProfileIdC = databaseBuilder.factory.buildTargetProfile().id;
       return databaseBuilder.commit();
     });
 
     it('should save all the target profile shares for the organization', async function() {
       // given
-      const targetProfileIdList = [targetProfileA.id, targetProfileB.id, targetProfileC.id];
+      const targetProfileIdList = [targetProfileIdA, targetProfileIdB, targetProfileIdC];
 
       // when
-      await targetProfileShareRepository.addToOrganization({ organizationId: organization.id, targetProfileIdList });
+      await targetProfileShareRepository.addTargetProfilesToOrganization({ organizationId, targetProfileIdList });
 
       // then
-      const targetProfileShares = await knex('target-profile-shares').where({ organizationId: organization.id });
+      const targetProfileShares = await knex('target-profile-shares').where({ organizationId });
       expect(targetProfileShares).to.have.lengthOf(3);
-      expect(_.map(targetProfileShares, 'targetProfileId')).to.have.members([targetProfileA.id, targetProfileB.id, targetProfileC.id]);
+      expect(_.map(targetProfileShares, 'targetProfileId')).to.have.members([targetProfileIdA, targetProfileIdB, targetProfileIdC]);
+    });
+
+    it('should not erase old target profil share', async function() {
+      // given
+      databaseBuilder.factory.buildTargetProfileShare({ organizationId, targetProfileId: targetProfileIdA });
+      await databaseBuilder.commit();
+      const targetProfileIdList = [targetProfileIdB, targetProfileIdC];
+
+      // when
+      await targetProfileShareRepository.addTargetProfilesToOrganization({ organizationId, targetProfileIdList });
+
+      // then
+      const targetProfileShares = await knex('target-profile-shares').where({ organizationId });
+      expect(targetProfileShares).to.have.lengthOf(3);
+      expect(_.map(targetProfileShares, 'targetProfileId')).to.have.members([targetProfileIdA, targetProfileIdB, targetProfileIdC]);
+    });
+
+    it('should throw a NotFound error when given target profile id does not exist', async function() {
+      // given
+      const targetProfileIdList = [999];
+
+      // when
+      const error = await catchErr(targetProfileShareRepository.addTargetProfilesToOrganization)({ organizationId, targetProfileIdList });
+
+      // then
+      expect(error).to.be.instanceOf(NotFoundError);
+      expect(error.message).to.equal('Le profil cible 999 n\'existe pas.');
+    });
+
+    it('should throw a NotFound error when given target profile id does not exist and not insert other existing target profiles', async function() {
```

Comment:
Petite proposition pour éviter les 'when' dans un it: rajouter un contexte. Ex: 
```javascript 
describe('when given target profile id does not exist', () => {
    it('should throw a NotFound error ', async function () { ... }
   describe('when given target profile not insert other existing target profiles', () => {
        it('it should throw a NotFound error too', async function() { .... }
} 
```
Après dans ce contexte là on a pas beaucoup de contextes différents donc je sais pas si les describes imbriqués sont pertinent mais on m'a dit d'éviter les "when" dans les "it".

Reply:
Il y a des philosophies différentes. D'un autre côté ça complexifie le code d'avoir une indentation de plus.

Reply:
Je dirai plutôt que la lecture est moins aisé 😃